src/pages/staticPages/webAPI/WebAPIPage.tsx

```diff
@@ -24,45 +24,19 @@ export class UserDataAccessToken {
     }
 }
 
-function buildDataAccessTokenFileContents(dat:UserDataAccessToken | undefined) {
-    if (!isNullOrUndefined(dat)) {
-        var token = dat!.token;
-        var creation_date = new Date(dat!.creationDate).toISOString();
-        var expiration_date = new Date(dat!.expirationDate).toISOString();
-        return `token: ${token}\ncreation_date: ${creation_date}\nexpiration_date: ${expiration_date}\n`;
-    } else {
-        alert("Cannot create Data Access Token file for user with non-existent tokens.");
-        return null;
-    }
-}
-
 @observer
 export default class WebAPIPage extends React.Component<{}, {}> {
     private get appStore(){
         return getBrowserWindow().globalStores.appStore;
     }
 
-    async generateNewDataAccessToken() {
-        if (this.appStore.isLoggedIn) {
-            let _token = await internalClient.createDataAccessTokenUsingPOST(
-                    {'allowRevocationOfOtherTokens':AppConfig.serverConfig.dat_uuid_revoke_other_tokens});
-            const dat = new UserDataAccessToken(_token.token, _token.creation, _token.expiration, _token.username);
-            return dat;
-        } else {
-            return undefined;
-        }
-    }
-
     async downloadDataAccessTokenFile() {
-        const dat = this.generateNewDataAccessToken();
-        if (!isNullOrUndefined(dat)) {
-            const fileContents = buildDataAccessTokenFileContents(await dat);
-            fileDownload(fileContents, "cbioportal_data_access_token.txt");
-        }
+        window.open("/api/data-access-token", "_blank");
```

Comment:
does this open data access token in a new window?   we want file download, yes?

Reply:
Yes, this opens a new tab. The backend sends a 'download file' response header to this tab. The browser downloads the DAT token and closes the tab.

Reply:
@pvannierop This doesn't work in all scenario's.  In my test case, I have a web app running in a local container accessible at the following url: 

> localhost:8080/cbioportal/

When I select the "Download Token" button, I get the error below - seems like the entire app path is not considered...

![image](https://user-images.githubusercontent.com/366003/77344907-400e4900-6d0a-11ea-821d-f842aa488a3d.png)